src/uwtools/tests/cli/test_templater.py

```diff
@@ -0,0 +1,283 @@
+    with patch.object(templater.sys, "argv", argv):
+        templater.main()
+    actual = capsys.readouterr().out.split("\n")
+    for line in expected.split("\n"):
+        assert line_in_lines(line, actual)
+
+
+@pytest.mark.parametrize("sw", [ns(d="-d", i="-i"), ns(d="--dry-run", i="--input-template")])
+def test_set_template_dry_run(sw, capsys):
+    """
+    Test dry-run output of ingest namelist tool.
+    """
+    infile = fixture_path("nml.IN")
+    expected = f"""
+Running with args:
+----------------------------------------------------------------------
+----------------------------------------------------------------------
+        outfile: None
+ input_template: {infile}
+    config_file: None
+   config_items: []
+        dry_run: True
+  values_needed: False
+        verbose: False
+          quiet: False
+Re-run settings: {sw.i} {infile} {sw.d}
+----------------------------------------------------------------------
+----------------------------------------------------------------------
+&salad
+  base = 'kale'
+  fruit = 'banana'
+  vegetable = 'tomato'
+  how_many = 22
+  dressing = 'balsamic'
+/
+""".lstrip()
+
+    with patch.dict(os.environ, {"fruit": "banana", "vegetable": "tomato", "how_many": "22"}):
+        argv = ["test", sw.i, infile, sw.d]
+        with patch.object(templater.sys, "argv", argv):
+            templater.main()
+        actual = capsys.readouterr().out.split("\n")
+        for line in expected.split("\n"):
+            assert line_in_lines(line, actual)
+
+
+@pytest.mark.parametrize("sw", [ns(i="-i"), ns(i="--input-template")])
+def test_set_template_listvalues(sw, capsys):
+    """
+    Test "values needed" output of ingest namelist tool.
+    """
+    infile = fixture_path("nml.IN")
+    expected = f"""
+Running with args:
+----------------------------------------------------------------------
+----------------------------------------------------------------------
+        outfile: None
+ input_template: {infile}
+    config_file: None
+   config_items: []
+        dry_run: False
+  values_needed: True
+        verbose: False
+          quiet: False
+Re-run settings: {sw.i} {infile} --values-needed
+----------------------------------------------------------------------
+----------------------------------------------------------------------
+Values needed for this template are:
+fruit
+how_many
+vegetable
+""".lstrip()
+    argv = ["test", sw.i, infile, "--values-needed"]
+    with patch.object(templater.sys, "argv", argv):
+        templater.main()
+    actual = capsys.readouterr().out.split("\n")
+    for line in expected.split("\n"):
+        assert line_in_lines(line, actual)
+
+
+@pytest.mark.parametrize(
+    "sw", [ns(d="-d", i="-i", v="-v"), ns(d="--dry-run", i="--input-template", v="--verbose")]
+)
+def test_set_template_verbosity(sw, capsys):
+    infile = fixture_path("nml.IN")
+    # #PM# WHAT TO DO ABOUT /dev/null BELOW?
```

Comment:
This relates to the default log file value being supplied in the `parse_args()` function -- I've set that to `/dev/null` to avoid having the logger create files in source-code directories, but we need to decide what to do about this in general.

Reply:
I'll just leave it for now. It will disappear if/when we revamp logging.